### PR TITLE
Add "Request type" item to be selected by user with PALGA role

### PR DESCRIPTION
**Pull request by ewelinagr**
[DNTP-630](https://jira.thehyve.nl/browse/DNTP-630)

Depends on https://github.com/thehyve/dntp-portal/pull/314.

### CODE REVIEW

File: src/main/java/business/services/RequestFormService.java [hunk 1]
```diff
@@ -390,6 +390,8 @@ public void transferData(HistoricProcessInstance instance, RequestRepresentation
                 request.setAgreementReached(fetchBooleanVariable("agreement_reached", variables));
                 request.setAgreementNotApplicable(fetchBooleanVariable("agreement_not_applicable", variables));
 
+                request.setRequestType((String)variables.get("request_type"));
```
Comment (gijskant) [inline src/main/java/business/services/RequestFormService.java]: I would prefer to store this value in the `RequestProperties`, as it is not required for the process.

File: src/main/resources/static/app/request/request-contents.html [hunk 2]
```diff
@@ -347,6 +347,20 @@ <h4 class="h4">{{ 'Letter of medical research ethics committee' | translate }}</
     </div>
 
     <div ng-if="isPalga()">
+
+        <!-- Request type -->
+        <h4 class="h4 h4-palga">{{'Request type'|translate}}</h4>
+        <hr>
```
Comment (gijskant) [inline src/main/resources/static/app/request/request-contents.html]: The `<hr>` is only here, which makes it inconsistent

File: src/main/resources/static/app/request/request-review.html [hunk 3]
```diff
@@ -1,4 +1,17 @@
 <div>
+  <h4 class="h4 h4-palga">{{'Request type'|translate}}</h4>
+  <hr>
+
+  <div class="row">
+    <div class="col-sm-6 form-control-static" >
+      <select class="form-control" name="requestType" id="requestType"
+              ng-model="request.requestType" ng-options="type | translate for type in requestTypes"
+              ng-required="true">
+        <option value="" ng-selected="selected">&ndash; {{'Select request type' | translate}} &ndash;</option>
```
Comment (gijskant) [inline src/main/resources/static/app/request/request-review.html]: Apparently not clearly mentioned in the ticket: when selecting 'Other', the user should be able to type a value (from the proposal document: 
> Overig → er opent een tekst veld waar  we zelf wat kunnen in typen

Comment (gijskant) [reply src/main/resources/static/app/request/request-review.html]: That custom value could be saved in the same property. You could check the specialism (in user registration) for an example.

File: src/main/resources/static/app/request/request-contents.html (hunk 2 above)
Comment (ewelinagr) [reply src/main/resources/static/app/request/request-contents.html]: oups, copy-paste from request-review...

File: src/main/resources/static/app/request/request-review.html (hunk 3 above)
Comment (ewelinagr) [reply src/main/resources/static/app/request/request-review.html]: it's updated now

**Comment by gijskant** [comment]
Looks like request type 'Other' does not yet open an input field:

Screenshot dntp-portal request type other

**Comment by gijskant** [comment]
Sorry, forgot to update branch.

File: src/main/resources/static/app/request/request-review.html [hunk 4]
```diff
@@ -1,4 +1,27 @@
 <div>
+  <h4 class="h4 h4-palga">{{'Request type'|translate}}</h4>
+  <hr>
+
+  <!-- Request type -->
+  <div class="form-group" ng-class="{ 'has-error' : registrationForm.requestType.$invalid}">
```
Comment (gijskant) [inline src/main/resources/static/app/request/request-review.html]: `registrationForm` should be `requestEditForm`

Comment (ewelinagr) [reply src/main/resources/static/app/request/request-review.html]: this fixed the issue with validation messages, thanks!

**Pull request merged**